src/libraries/System.Diagnostics.StackTrace/tests/StackTraceTests.cs

```diff
@@ -300,6 +313,52 @@ public void ToString_NullFrame_ThrowsNullReferenceException()
             Assert.Equal(Environment.NewLine, stackTrace.ToString());
         }
 
+        [Fact]
+        public void ToString_ShowILOffset()
+        {
+            // Normal loading case
+            var asm = Assembly.LoadFrom(SourceTestAssemblyPath);
+            try
+            {
+                asm.GetType("Program").GetMethod("Foo").Invoke(null, null);
+            }
+            catch (Exception e)
+            {
+                Assert.Contains(":token ", e.InnerException.StackTrace);
+            }
+
+            // Assembly.Load(Byte[]) case
+            var inMemBlob = File.ReadAllBytes(SourceTestAssembly2Path);
+            var asm2 = Assembly.Load(inMemBlob);
+            try
+            {
+                asm2.GetType("Program2").GetMethod("Foo").Invoke(null, null);
+            }
+            catch (Exception e)
+            {
+                Assert.Contains(":token ", e.InnerException.StackTrace);
+            }
+
+            // AssmblyBuilder.DefineDynamicAssembly() case
+            AssemblyName asmName = new AssemblyName("ExceptionTestAssembly3");
+            AssemblyBuilder asmBldr = AssemblyBuilder.DefineDynamicAssembly(asmName, AssemblyBuilderAccess.Run);
+            ModuleBuilder modBldr = asmBldr.DefineDynamicModule(asmName.Name);
+            TypeBuilder tBldr = modBldr.DefineType("Program3");
+            MethodBuilder mBldr = tBldr.DefineMethod("Foo", MethodAttributes.Public | MethodAttributes.Static, null, null);
+            ILGenerator ilGen = mBldr.GetILGenerator();
+            ilGen.ThrowException(typeof(NullReferenceException));
+            ilGen.Emit(OpCodes.Ret);
+            Type t = tBldr.CreateType();
+            try
+            {
+                t.InvokeMember("Foo", BindingFlags.InvokeMethod, null, null, null);
+            }
+            catch (Exception e)
+            {
+                Assert.Contains(":token ", e.InnerException.StackTrace);
```

Comment:
Should we assert that there is at least one frame (in this case) that does have a source path?

Reply:
An assert was added to check whether this functionality works in the StackTrace of the exception that occurred in the method of TestAssembly without pdb.